shared/fs/really-delete/container.js

```diff
@@ -0,0 +1,47 @@
+// @flow
+import * as FsGen from '../../actions/fs-gen'
+import * as RouteTreeGen from '../../actions/route-tree-gen'
+import * as Container from '../../util/container'
+import ReallyDelete from '.'
+import {anyWaiting} from '../../constants/waiting'
+import * as Types from '../../constants/types/fs'
+import * as Constants from '../../constants/fs'
+
+type OwnProps = Container.RouteProps<{path: Types.Path}, {}>
+
+const mapStateToProps = state => {
+  return {
+    _deleting: (path: Types.Path) => anyWaiting(state, Constants.deleteFolderWaitingKey(path)),
+    title: 'Confirmation',
+  }
+}
+
+const mapDispatchToProps = dispatch => ({
+  _onDelete: (path: Types.Path) => {
+    dispatch(FsGen.createDeleteFile({path}))
+    // TODO: I think I should be closing this using the waiting key instead,
+    //  but I'm not sure how that works.
```

Comment:
You could fire `navigateUp` in `componentDidUpdate` when waiting state clears. See `TriggerActionsWhenNeeded` in send-link-to-chat. It doesn't use the waiting key, but the logic should be similar.

Reply:
I've changed it to do this.